examples/scripts/deconv_modl_train_foam.py

```diff
@@ -0,0 +1,348 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This file is part of the SCICO package. Details of the copyright
+# and user license can be found in the 'LICENSE.txt' file distributed
+# with the package.
+
+r"""
+Training of MoDL for Deconvolution
+==================================
+
+This example demonstrates the training and application of a model-based deep learning
+(MoDL) architecture described in :cite:`aggarwal-2019-modl`
+ for a deconvolution (deblurring) problem.
+
+The source images are foam phantoms generated with xdesign.
+
+A class [flax.MoDLNet](../_autosummary/scico.learning.rst#scico.learning.MoDL)
+ implements the MoDL architecture, which solves the optimization problem
+
+  $$\mathrm{argmin}_{\mathbf{x}} \; \| A \mathbf{x} - \mathbf{y} \|_2^2 + \lambda \, \| \mathbf{x} - \mathrm{D}_w(\mathbf{x})\|_2^2 \;,$$
```

Comment:
Include (1/2) for l2 norm?

Reply:
I am using directly the formulation from the MoDL reference.